web_responsive/static/src/legacy/js/web_responsive.js

```diff
@@ -146,6 +146,7 @@ odoo.define("web_responsive", function (require) {
             const options = this._super.apply(this, arguments);
             _.extend(options, {
                 on_clear: () => this.reinitialize(false),
+                domain: this.record.getDomain(this.recordParams),
```

Comment:
Maybe this should be a fix to Odoo itself, but maybe it's hard to reproduce in core...

Reply:
The only way I found to reproduce it in core is by adding a many2one field in a line model like "sale.order.line" adding a domain in the form view of the line, and in mobile view pressing the button search more

Reply:
Well, that's precisely the case, isn't it? :sweat_smile: